Document the 'wal_cleanup_delay' option (#3350)
tarantool/doc: doc/reference/configuration/cfg_binary_logging_snapshots.rst

    | Default: 16777216 bytes
    | Environment variable: TT_WAL_QUEUE_MAX_SIZE
    | Dynamic: **yes**

.. _cfg_binary_logging_snapshots-wal_cleanup_delay:

.. confval:: wal_cleanup_delay

>     Since version :doc:`2.8.1 </release/2.8.1>`.

Reviewer: The option already appeared in 2.6.3 (2.6.2-148-gd129729d9) and 2.7.2 (2.7.1-152-ge8c094644) (we had rolling releases back then, so 2.8.1 was released together with 2.7.2. and 2.6.3). This was considered a bugfix, so it was cherry-picked from 2.8.1 to the older versions.

    The delay (in seconds) used to prevent the :ref:`Tarantool garbage collector <cfg_checkpoint_daemon-garbage-collector>`
    from immediate removing :ref:`write-ahead log<internals-wal>` files after a node restart.

>     This delay helps :ref:`replicas <replication-roles>` sync with a master faster after its restart and

Reviewer: I wouldn't say it helps to sync faster: it eliminates the risk that master deletes WALs needed for replicas after a restart. It fixes a possible erroneous situation rather than improves anything.

Although, yes, this means replicas do not risk downloading the data all over again, and thus can sync faster.

    | Default: 16777216 bytes
    | Environment variable: TT_WAL_QUEUE_MAX_SIZE
    | Dynamic: **yes**

.. _cfg_binary_logging_snapshots-wal_cleanup_delay:

.. confval:: wal_cleanup_delay

    Since version :doc:`2.6.3 </release/2.6.3>`.
    The delay (in seconds) used to prevent the :ref:`Tarantool garbage collector <cfg_checkpoint_daemon-garbage-collector>`
    from immediately removing :ref:`write-ahead log<internals-wal>` files after a node restart.
    This delay eliminates possible erroneous situations when the master deletes WALs
    needed by :ref:`replicas <replication-roles>` after restart.

>     As a consequence, replicas sync with a master faster after its restart and

Reviewer 2: the master?

Author: sure, thanks

    don't need to download all the data again.

    Once all the nodes in the replica set are up and running,
    automatic cleanup is started again even if ``wal_cleanup_delay`` has not expired.

    .. NOTE::

>         The ``wal_cleanup_delay`` option is not in effect if a node is running as an

Reviewer 2: If I get it right , this wording will be more precise:

```suggestion
        The ``wal_cleanup_delay`` option has no effect on nodes running as
```

I mean, it still has effect on other nodes in the replica set except anonymous replicas, right?

Author: Thanks, sounds better.
